version: '3.8'

services:
  app:
    build:
      context: ./docker/app
    container_name: peligro-laravel-app
    working_dir: /var/www/html
    volumes:
      - ./laravel:/var/www/html
    environment:
      - APP_ENV=local
      - APP_DEBUG=true
      - APP_KEY=base64:K0QkoJdjC83TX5sJ76/O1XLzp5nN+G6XG8uhxdjBzQM=
    ports:
      - "8000:8000"
    networks:
      - custom_network
    

  nginx:
    image: nginx:alpine
    container_name: laravel-nginx
    volumes:
      - ./laravel:/var/www/html
      - ./docker/nginx/default.conf:/etc/nginx/conf.d/default.conf
    ports:
      - "8080:80"
    depends_on:
      - app
    networks:
      - custom_network

  python:
    build:
      context: ./docker/python
    container_name: python_service
    volumes:
      - ./python:/usr/src/app
    working_dir: /usr/src/app
    command: tail -f /dev/null  # Mantener el contenedor en ejecuci√≥n
    ports:
      - "5000:5000"
    environment:
      - PYTHONUNBUFFERED=1
    networks:
      - custom_network

networks:
  custom_network:
    driver: bridge
    ipam:
      config:
        - subnet: 172.42.12.0/28
          gateway: 172.42.12.1
